### PR TITLE
small fixes and bump minimum GCC version to 4.9.1

### PR DESCRIPTION
**Pull request by carmichaelong**
@tgeijten Let me know if you want any changes before you merge.

### CODE REVIEW

File: flut/system_tools.cpp [hunk 1]
```diff
@@ -71,13 +71,25 @@ namespace flut
...
-		auto in_time_t = std::chrono::system_clock::to_time_t( std::chrono::system_clock::now() );
+        // GCC did not implement std::put_time until GCC 5
+#if defined(__GNUC__) && (__GNUC__ < 5)
+        auto in_time_t = std::chrono::system_clock::to_time_t( std::chrono::system_clock::now() );
+        char arr[100];
+
+        std::stringstream ss;
+        ss << std::strftime(arr, sizeof(arr), format, std::localtime(&in_time_t));
+        return ss.str();
+#else
```
Comment (tgeijten) [inline flut/system_tools.cpp:85]: Doesn't the result get stored in `arr`? If so, you can directly create a string from that and omit the `stringstream`.

**Review by tgeijten — CHANGES_REQUESTED** [review]

**Comment by carmichaelong** [comment]
You're definitely right. Should be fixed now.

File: flut/system_tools.cpp [hunk 2]
```diff
@@ -77,10 +77,10 @@ namespace flut
...
 #if defined(__GNUC__) && (__GNUC__ < 5)
 		auto in_time_t = std::chrono::system_clock::to_time_t( std::chrono::system_clock::now() );
-		char arr[100];
 
-		std::strftime(arr, sizeof(arr), format, std::localtime(&in_time_t));
-		string str(arr);
+		char arr[100];
+		std::strftime( arr, sizeof( arr ), format, std::localtime( &in_time_t ) );
+		string str( arr );
 		return str;
```
Comment (tgeijten) [inline flut/system_tools.cpp:84]: Now that we're nitpicking... ;-) Just write `return string( arr )`. It's shorter and immediately clear that the returned value is an rvalue.

**Review by tgeijten — APPROVED** [review]

File: flut/system_tools.cpp (hunk 2 above)
Comment (carmichaelong) [reply flut/system_tools.cpp:84]: I appreciate the nitpicking! I still have much C++ to learn... Thanks for the merge!

**Pull request merged**